File: Array&String/CPP/count_good_triplets.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
    public:
        int countGoodTriplets(vector<int>& arr, int a, int b, int c) {
            int cnt = 0, n = arr.size();
            for (int i = 0; i < n - 2; i++) {
                for (int j = i + 1; j < n - 1; j++) {
                    int a_ = abs(arr[i] - arr[j]);
                    if (a < a_) continue;
                    for (int k = j + 1; k < n; k++) {
                        int b_ = abs(arr[j] - arr[k]), c_ = abs(arr[i] - arr[k]);
                        if (b < b_ || c < c_) continue;
                        if (a_ <= a && b_ <= b && c_ <= c) cnt++;
                    }
                }
            }
            return cnt;
        }
    };

File: Array&String/count_good_triplets.py
class Solution:
    def countGoodTriplets(self, arr: list[int], a: int, b: int, c: int) -> int:
        cnt, n = 0, len(arr)
        for i in range(n-2):
            for j in range(i+1, n-1):
                a_ = abs(arr[i] - arr[j])
                if a_ > a: continue
                for k in range(j+1, n):
                    b_, c_ =  abs(arr[j] - arr[k]), abs(arr[i] - arr[k])
                    if b_ > b or c_ > c: continue
                    if a_ <= a and b_ <= b and c_ <= c: cnt += 1
        return cnt